crates/pubsub/src/service.rs

```diff
@@ -89,15 +91,13 @@ where
         // Re-subscribe to all active subscriptions
         tracing::debug!(count = self.subs.len(), "Re-starting active subscriptions");
 
-        // Drop all server IDs. We'll re-insert them as we get responses.
-        self.subs.drop_server_ids();
```

Comment:
removing this causes memory leak on each reconnect

Reply:
server IDs are dropped as they are no longer valid, btw. the server will issue new IDs on reconnect and all old IDs are now dead

Reply:
I remember that new subscription messages were not sent to the frontend if server ids weren't dropped... I'll do more tests

Reply:
Ok, so here's what's happening:

On reconnect server ID are dropped (i.e. `local_to_server` is cleared). When the new subscription response is handled (in [handle_sub_response](https://github.com/alloy-rs/alloy/blob/d68e1103bda0a0b95e1928c56091dc48e64c95de/crates/pubsub/src/service.rs#L179)), we call `upsert` to insert or update the server_id. 

But, as `local_to_server` is cleared, instead of just changing the server ID, we insert a new `ActiveSubscription`, dropping the old one *and* the associated `Sender`, leading to close the corresponding `Receiver` on the frontend side.